source/mainnet/smart-contracts/guides/setup-contract.rst

```diff
@@ -29,6 +29,27 @@ This is done by adding the following in the ``Cargo.toml`` file ::
    [lib]
    crate-type = ["cdylib", "rlib"]
 
+
+Starting a smart contract project from a template
```

Comment:
I think this section is fine and clear, but I think we should change the order of the document a bit.
At least, we shouldn't split up the `cargo init` and "add dependency" sections, since they belong together.

I would propose something like:

- Intro:
  - There are two approaches: from template or from scratch.
- From template:
   - ...
- From scratch
   - cargo init
   - add dependency

Reply:
I would recommend newcomers to use a template, so I also think that we should list it before the scratch approach.